projects/clr/hipamd/src/hip_graph_internal.cpp

```diff
@@ -539,18 +541,28 @@ bool Graph::RunOneNode(Node node, bool wait) {
     // Create a wait list from the last launches of all dependencies
     for (auto dep : wait_order_) {
       if (dep != nullptr) {
-        // Add all commands in the wait list
-        if (dep->GetType() != hipGraphNodeTypeGraph) {
-          for (auto command : dep->GetCommands()) {
-            waitList.push_back(command);
-          }
+        for (auto command : dep->GetCommands()) {
+          waitList.push_back(command);
         }
       }
     }
     if (node->GetType() == hipGraphNodeTypeGraph) {
       // Process child graph separately, since, there is no connection
       auto child = reinterpret_cast<hip::ChildGraphNode*>(node)->GetChildGraph();
       if (!reinterpret_cast<hip::ChildGraphNode*>(node)->GetGraphCaptureStatus()) {
+        // Debug print for waitList before running child graph
+        if (!waitList.empty()) {
+          ClPrint(amd::LOG_INFO, amd::LOG_CODE,
```

Comment:
LOG_CODE is for CO prints. It has nothing to do with the Graph functionality. Add LOG_GRAPH